crates/autopilot/src/infra/persistence/api/mod.rs

```diff
@@ -0,0 +1,4 @@
+pub mod auction;
+pub mod competition;
+pub mod events;
+pub mod quote;
```

Comment:
What is the reasoning behind the api submodule and this structuring? (does it contain infra code that is used by the API, which API specifically)?

I'm not a fan of structuring the modules this way. Not having a single domain facing persistence modules makes it very hard to see what persistence the autopilot requires (I always struggle navigating our db queries because they are scattered across many different files). 
If the top level `Persistence` module becomes to big, I think we should turn it into a lightweight facade that dispatches the work into submodules that are organized by the type of persistence that is being used (e.g. persistence::s3 vs. persistence::postgres). If then the postgres module is also becoming too big, maybe we can think about extensions that are named after the type of queries they perform.

Reply:
refactored.

Reply:
I see that this will likely become too big if we move all db queries. In this case, I'd still have one domain facing struct `Persistence` which just dispatches the calls to sub-persistence solutions (postgres, s3, etc) and use logically sepearated modules within each persistence type.

For now I think the refactored version is ok.